handlers/retest/ci.go

```diff
@@ -0,0 +1,225 @@
+	TravisCIEndPoint = "https://api.travis-ci.org" //travis-ci endpoint
+	ContentTypeJSON  = "application/json"          //content-type
+	TravisAPIVersion = "Travis-API-Version"        //API version, a Mandatory header field for Travis-CI V3 APIs call
+)
+
+type TravisJobType string
+
+//Travis Build Job Names
+const (
+	Build       TravisJobType = "build"
+	Verify      TravisJobType = "verify"
+	Unittest    TravisJobType = "unittest"
+	Integration TravisJobType = "integration"
+	Crossbuild  TravisJobType = "crossbuild"
+)
+//Travis Job Id's
+const(
+	JobBuild        = iota + 1
+	JobVerify
+	JobUnittest
+	JobIntegration
+	JobCrossbuild
+)
+
+//GetJobIdsFromTravisBuild Function to handle Get Jobs from TravisCI
+func GetJobIdsFromTravisBuild(BuildRefId string, Token string)(error, []byte) {
+	var Reqbody io.Reader
+
+	url := fmt.Sprintf("%s%s/jobs", TravisCIEndPoint, BuildRefId)
+	req, err := http.NewRequest(http.MethodGet, url, Reqbody)
+	if err != nil {
+		glog.Errorf("HTTP Request failed: %v", err)
+		return err,[]byte("")
+	}
+	err, _, Body := SendHttpReqToCI(req, Token)
+	return nil,Body
+}
+
+//TriggerJob Function to handle Triggering Jobs build in TravisCI
+func TriggerJob(JobBuildId string, Token string) (error, int) {
+	var Reqbody io.Reader
+	var statusCode int
+	url := fmt.Sprintf("%s%s/restart", TravisCIEndPoint, JobBuildId)
+	req, err := http.NewRequest(http.MethodPost, url, Reqbody)
+	if err != nil {
+		glog.Errorf("HTTP request failed: %v", err)
+		return err, statusCode
+	}
+	err, statusCode, _ = SendHttpReqToCI(req, Token)
+	if statusCode == http.StatusAccepted {
+		glog.Info("Restart Job Build is successfully triggered !!")
+	} else {
+		glog.Infof("Restart Job Build is failed to trigger, HttpStatus Code: %v" , http.StatusAccepted)
+		return nil, statusCode
+	}
+	return nil, statusCode
+}
+
+//SendHttpReqToCI Function to handle send HTTP request to TravisCI
+func SendHttpReqToCI(req *http.Request, Token string) (error, int, []byte) {
+	client := &http.Client{}
+
+	req.Header.Set("Content-Type", ContentTypeJSON)
+	req.Header.Set("Authorization", "token "+Token)
+	req.Header.Set(TravisAPIVersion, "3")
+	//Http Do req, For getting the PR Request information for particular repo
+	resp, err := client.Do(req)
+	if err != nil {
+		glog.Errorf("HTTP Do request failed: %v", err)
+		return nil, resp.StatusCode, []byte("")
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		glog.Errorf("Failed to read resp: %v", err)
+	}
+	return err, resp.StatusCode, body
+}
+//StartToTriggerJob Function to handle travis build job trigger
+func StartToTriggerJob(TravisJobRespBody TravisJobRespStruct, jobname TravisJobType, token string) error {
+	var err error
+	//Range over all jobs to match the JonID and Trigger Build
+	for _, job := range TravisJobRespBody.Jobs {
+		JobId := strings.Split(job.Number, ".")[1]
+		BuildJobId, _ := strconv.Atoi(JobId)
+		if Build == jobname && BuildJobId == JobBuild {
+			err, _ = TriggerJob(job.Href, token)
```

Comment:
`
There are some duplicated code blocks below:

```
err, _ = TriggerJob(job.Href, token)
if err != nil {
	glog.Errorf("Failed to Trigger Job: %v", err)
}
break
```
Can you resolve the duplications?

Reply:
@m1093782566 , im just calling the TriggerJob function, based on the Job Name, its not a duplicate code, as we have different jobs in a build, each job has to call the TriggerJob function for trigering the Job.

Reply:
To be clearer, can't we do something like: 

```
if if Build == jobname && BuildJobId == JobBuild ||  Verify == jobname && BuildJobId == JobVerify ... {
err, _ = TriggerJob(job.Href, token)
if err != nil {
	glog.Errorf("Failed to Trigger Job: %v", err)
}
break
}
```

OR even just:

```
if Consistent(jobname, BuildJobId) {
err, _ = TriggerJob(job.Href, token)
if err != nil {
	glog.Errorf("Failed to Trigger Job: %v", err)
}
break
}
```

??